src/main/java/de/tum/cit/aet/artemis/exercise/repository/ExerciseRepository.java

```diff
@@ -322,6 +322,9 @@ AND EXISTS (
             """)
     Set<String> findAllCategoryNames(@Param("courseId") Long courseId);
 
+    @EntityGraph(attributePaths = "categories")
+    List<Exercise> findAllWithCategoriesByCourseId(Long courseId);
```

Comment:
if this is only used in tests, I'd prefer to move it into a test repository

Reply:
Should I create a test repo in src/test/java/de/tum/cit/aet/artemis/exercise or do we have a general test repo already?